sql/src/test/java/org/apache/druid/sql/calcite/expression/ExpressionsTest.java

```diff
@@ -1779,8 +1778,7 @@ public void testTimeMinusYearMonthInterval()
             DruidExpression.functionCall("timestamp_shift"),
             ImmutableList.of(
                 DruidExpression.ofColumn(ColumnType.LONG, "t"),
-                // RexNode type "interval year to month" is not reported as ColumnType.STRING
-                DruidExpression.ofLiteral(null, DruidExpression.stringLiteral("P13M")),
+                DruidExpression.ofLiteral(ColumnType.LONG, DruidExpression.stringLiteral("P13M")),
```

Comment:
Long type for string literal? Seems not right.

Reply:
hmm, so this was a result of merging the rules from cast into `Calcites.getColumnTypeForRelDataType` which had interval types as LONG, but agree it does look a bit strange so will fix to report these interval types as strings instead and see what that breaks

Reply:
ah, i see why it was treating the cast as LONG typed...

`timestampDiff` test for example starts spitting out explicit casts which are not necessary so I think i need to add a bit extra logic to cast to let intervals cast to longs without a cast ```ExpressionVirtualColumn{name='v0', expression='div(CAST(("__time" - 915148800000), 'LONG'),86400000)', outputType=LONG}```

Reply:
ok, updated to handle intervals as STRINGS so they match the argument literals in plans, but also handle them as longs for casting (and also reduction). I suppose instead I could just drop intervals from being handled by `Calcites.getColumnTypeForRelDataType` like it was before, but string seems the most correct i think?